src/main/java/com/redhat/devtools/lsp4ij/features/formatting/LSPEditorImprovementsTypedHandler.java

```diff
@@ -73,9 +75,10 @@ private boolean handleNestedQuote(@NotNull PsiFile file,
         String stringLiteral = getStringLiteral(file, documentChars, offset);
         if (StringUtil.isNotEmpty(stringLiteral)) {
             char stringLiteralQuote = stringLiteral.charAt(0);
-            // Single quote within double-quoted string or vice-versa
-            // Escaped single quote within single-quoted string or vice-versa
-            if ((stringLiteralQuote != quote) || ((offset > 0) && (documentChars.charAt(offset - 1) == '\\'))) {
+            // Quote character within a string bounded by another quote character
```

Comment:
Formatting changes and comment improvements while I was already in this file. No logic changes at all.